packages/react/src/HeroImage/HeroImage.tsx

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright (c) 2023 Gemeente Amsterdam
+ */
+
+import clsx from 'clsx'
+import { forwardRef, type ImgHTMLAttributes, type PropsWithChildren, type ReactNode } from 'react'
+import { Image } from '../Image/Image'
+
+export interface HeroImageProps extends PropsWithChildren<ImgHTMLAttributes<HTMLImageElement>> {
+  /*
+   * A custom image component to use instead of the one from this library.
+   * Note that the image must fill the entire content box, e.g. through `object-fit: cover`.
+   */
+  ImageComponent?: ReactNode
+}
+
+export const HeroImage = forwardRef<HTMLImageElement, HeroImageProps>(
+  ({ children, className, ImageComponent, ...restProps }: HeroImageProps, ref) => (
+    <div className={clsx('amsterdam-hero-image', className)}>
+      <div className="amsterdam-hero-image__image">
+        {ImageComponent ?? <Image {...restProps} alt="" objectFit="cover" ref={ref} />}
+      </div>
+      {children}
+      <HeroImage src="abc.jpg" />
```

Comment:
This is probably not intentional, right? This makes it a recursive function :D

Reply:
Oh wow, I must have been typing in the wrong window there 😂 How did this not fail a test? Removed.